mindsdb_server/namespaces/datasource.py

```diff
@@ -66,11 +67,108 @@ def save_datasource_metadata(ds):
         with open(os.path.join(mindsdb.CONFIG.MINDSDB_DATASOURCES_PATH, ds['name'], 'metadata.json'), 'w') as fp:
             json.dump(ds, fp)
 
-def create_sqlite_db(path, ds):
+def create_sqlite_db(path, data_frame):
     con = sqlite3.connect(path)
-    ds.to_sql(name='data', con=con, index=False)
+    data_frame.to_sql(name='data', con=con, index=False)
     con.close()
 
+def get_analysis(source):
+    global global_mdb
+    return global_mdb.analyse_dataset(source, sample_margin_of_error=0.025)
+
+def cast_df_columns_types(df):
+    types_map = {
+        'Numeric': {
+            'Int': 'int64',
+            'Float': 'float64',
+            'Binary': 'bool'
+        },
+        'Date': {
+            'Date': 'datetime64',       # YYYY-MM-DD
+            'Timestamp': 'datetime64'   # YYYY-MM-DD hh:mm:ss or 1852362464
+        },
+        'Categorical': {
+            'Binary Category': 'category',
+            'Category': 'category'
+        },
+        'File Path': {
+            'Image': 'object',
+            'Video': 'object',
+            'Audio': 'object'
+        },
+        'Url': {},
+        'Sequential': {
+            'Text': 'object',
+            'Array': 'object'
+        }
+    }
+
+    analysis = get_analysis(df)
+    columns = [dict(name=x) for x in list(df.keys())]
+
+    for column in columns:
+        try:
+            name = column['name']
+            col_type = analysis['data_analysis_v2'][name]['typing']['data_type']
+            col_subtype = analysis['data_analysis_v2'][name]['typing']['data_subtype']
+            new_type = types_map[col_type][col_subtype]
+            if new_type == 'int64' or new_type == 'float64':
+                df[name] = df[name].apply(lambda x: x.replace(',','.'))
+            if new_type == 'int64':
+                df = df.astype({name: 'float64'})
+            df = df.astype({name: new_type})
+        except Exception as e:
+            print(e)
+            print(f'Error: cant convert type of DS column {name} to {new_type}')
+    
+    return df
+
+def parse_filter(key, value):
+    result = re.search(r'filter(_*.*)\[(.*)\]', key)
+    operator = result.groups()[0].strip('_') or 'like'
+    field = result.groups()[1]
+    operators_map = {
+        'like': 'like',
+        'in': 'in',
+        'nin': 'not in',
+        'gt': '>',
+        'lt': '<',
+        'gte': '>=',
+        'lte': '<=',
+        'eq': '=',
+        'neq': '!='
+    }
+    if operator not in operators_map:
+        return None
+    operator = operators_map[operator]
+    return {'field': field, 'value': value, 'operator': operator}
+
+
+def prepare_sql_where(where):
+    marks = {}
+    if len(where) > 0:
+        for i in range(len(where)):
+            field = where[i]['field'].replace('"', '""')
```

Comment:
Should the `"` escaping be done on the value rather than the field?

Reply:
Fields inserted directly to query string, so they need to be escaped.
Values escaped by sqlite3 driver. I`ve tried to make some sql-injection to value without any result.